### PR TITLE
iox-#1345 do not create logstream object when logging is off

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PRs removes the logger from the compiled binary when `IOX_MINIMAL_LOG_LEVEL == OFF`.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1345

### CODE REVIEW

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/log/logstream.inl [hunk 1]
```diff
@@ -200,6 +200,27 @@ inline LogStream& LogStream::operator<<(const LogLevel value) noexcept
     return *this;
 }
 
+namespace internal
```
Comment (mossmaurice) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/log/logstream.inl:203]: I thought you like

```suggestion
namespace detail
```

more :grin:

**Review by mossmaurice — APPROVED** [review]
LGTM, thanks!

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/log/logstream.inl (hunk 1 above)
Comment (elBoberido) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/log/logstream.inl:203]: Hahaha. Good point but only after it is moved to the new location :)

If another commit is required I'll do it with this PR.

Comment (elfenpiff) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/log/logstream.inl:203]: Hey, it is me the party pooper :smile: 

I my opinion we gain little to nothing with introducing `detail` as internal namespace. 
But we have to pay for it with an non insignificant amount of work.

Don't get me wrong here, I am not against it but could we start with this when we are once bored again and are looking for something to do?

**Pull request merged**